test/lib/flat_mutation_reader_assertions.hh

```diff
@@ -736,23 +773,45 @@ public:
         return *this;
     }
 
-    flat_reader_assertions_v2& produces_range_tombstone_change(const range_tombstone_change& rt) {
-        testlog.trace("Expect {}", rt);
-        auto mfo = read_next();
-        if (!mfo) {
-            BOOST_FAIL(format("Expected range tombstone {}, but got end of stream", rt));
+    flat_reader_assertions_v2& may_produce_tombstones(position_range range) {
+        testlog.trace("Expect possible range tombstone changes in {}", range);
+        while (auto next = peek_next()) {
+            if (!next->is_range_tombstone_change()) {
+                break;
+            }
+            auto rtc = maybe_drop_deletion_time(next->as_range_tombstone_change());
+            if (!interval<position_in_partition>{range.start(), range.end()}.contains(rtc.position(), position_in_partition::tri_compare{*_reader.schema()})) {
```

Comment:
I now realize that `range` is of type `position_range`. Note that `position_range` is of the form `[a, b)` by definition, so there is a reason it doesn't have an all-inclusive `contains()` method. Do tests fail if you use `position_range::contains()`? I guess range tombstone changes can be emitted with position == `b`, instead of < `b`.

Reply:
> I guess range tombstone changes can be emitted with position == b, instead of < b

exactly that, yes